addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -279,11 +280,16 @@ def _get_bom_data(self, bom, warehouse, product=False, line_qty=False, bom_line=
             bom_report_line['bom_cost'] += component['bom_cost']
         for component in components:
             if component['is_storable']:
-                if missing_qty := max(component['quantity'] - component['quantity_forecasted'], 0):
-                    missing_qty = float_repr(missing_qty, self.env['decimal.precision'].precision_get('Product Unit'))
-                    route_name = component['route_name'] or _('Order')
-                    component['status'] = _("%(qty)s To %(route)s", qty=missing_qty, route=route_name)
+                if missing_qty := max(component['quantity'] - component['quantity_available'], 0) and component['route_name'] == 'Order':  # Only kits use this route
+                    component['status'] = self.env._("%(qty)s To Order", qty=self._format_number_display(missing_qty), route=component['route_name'])
+                    component['missing_qty'] = missing_qty
+                elif missing_qty := max(component['quantity'] - component['quantity_forecasted'], 0):
+                    if component['route_name']:
+                        component['status'] = self.env._("%(qty)s To %(route)s", qty=self._format_number_display(missing_qty), route=component['route_name'])
+                        component['missing_qty'] = missing_qty
```

Comment:
If `component['route_name']` is "Order", then why would we need an extra condition?